### ISSUE TITLE
Ftp Adapter passive mode

### ISSUE THREAD

**Issue by costafrancesco94** [Digital-Peak/DPMedia-Free#1]
Hello :)

I was playing around with new Joomla4, and thought to try a custom filesystems driver to see how content integrate that.

I set up a sample connection with FTP adapter, pointing to a cPanel hosting I manage.

After some debug, I finally manage to connect and list files on that source, by activating _passive mode_.

I don't know implementation details of such FTP server, but I suppose without passive mode things are not working.

I leave you the "hotfix" I used to connect

File: `plg_filesystem_dpftp/src/Adapter/FtpAdapter.php`
```php

// ...... begin of file

protected function fetchFiles(string $path = '/'): array
	{
		$path = $this->getPath($path) ?: '/';

		$ftpPath = $path;
		if (strpos($path, '.') !== false) {
			$ftpPath = dirname($path);
		}

		$this->connect();

// BEGIN PATCH =============
		$this->getFtpClient()->pasv(true);
// END PATCH ===============

		$files = $this->getFtpClient()->mlsd($ftpPath);

// ...... rest of file

}
```

If this is usefoul, you may think to implement a field in config to activate passive mode on request....

Are PR welcome?

**Comment by laoneo** [comment]
Yes sure are pr's welcome. Would be nice if you can make it an option then. Thanks.

**Comment by laoneo** [comment]
Merged with #2.
